Automattic/vip-go-mu-plugins
# Clear old_slug cache when a post’s date is changed

## Description

Fixes https://github.com/Automattic/vip-go-mu-plugins/issues/1279

Currently, a post can get stuck in a redirect loop if the post's date is updated more than once. To resolve this, we need to also clear the `'old_slug' . $post->post_name` cache when the date is changed in case the site is displaying the publish date in the URL. Currently we assume only the post-slug being changed can affect this, which isn't true.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

See https://github.com/Automattic/vip-go-mu-plugins/issues/1279

Example:

1. Create a post on a site that includes the publish date in the permalink.
2. Update the post's publish date to a previous day. View the previous permalink and you should be redirected.
3. Update the date again, and view one of the previous permalinks, note that you are now stuck in a redirect loop.
4. Apply this patch, and try the above again. Now all previous dates/links will be properly redirected.

91623-zd
